### ISSUE TITLE
muted autoplay should display captions if available

### ISSUE THREAD

**Issue by aghassemi** [ampproject/amphtml#6376]
~~Should it? This is a discussion issue.~~

- [ ] Implement for amp-youtube
- [ ] Enable autoplay for Paul's video on https://www.ampproject.org/learn/about-amp/ 
- [ ] Implement for amp-video
Because we only autoplay videos muted, I argue we should display the video captions during autoplay and hide them after user taps a video to unmute it.

Since muted autoplay became popular, the number of videos with Text in the has increased `<random-number>%!` This is not good, but an indication that users may prefer to watch the entire video muted. So why don't we use captions by default to satisfy that use-case? 1- It solves the muted viewing problem, 2- Encourages video producers to create/enable captions which is a good thing. Many YouTube videos have captions anyway.

(Of course, we will have an opt-out option for this behaviour but will be opt-in by default)

Here are two demos, one with captions enabled by default and one without:
https://amphtml-ae5fc.firebaseapp.com/caption/test/manual/empty.amp.html (with autoplay+captions behavior)
https://amphtml-ae5fc.firebaseapp.com/caption/test/manual/empty-1.amp.html (without)

@rudygalfi @cramforce @edlea-g thoughts?
@mkhatib I think your thoughts are clear :) this issue was inspired by your tweets: https://twitter.com/mkhatib7/status/799765119446097921 and https://twitter.com/mkhatib7/status/799765587488477184

**Comment by cramforce** [comment]
I think this is a good idea to some extend, but I also strongly feel that @mkhatib is wrong about autoplay :)

We can't just turn on captions, because "produced for muted autoplay" wants its inline captions to be shown and not obfuscated by the system captions.

Having said this, I think we should definitely make it very easy to turn on captions by default during muted autoplay.

**Comment by aghassemi** [comment]
@cramforce So are you suggesting an opt-in option? I am hoping for an opt-out one. I think it is easier for the "produced for muted autoplay" author to turn it off than for everyone else to turn it on (which `<random number>%` of them won't do because they don't know about it and the user suffers at the end)
(If we go with opt-out, we should put an experiment flag on it)

**Comment by aghassemi** [comment]
if we go with opt-in, maybe something like `autoplay="withCaptions"`

**Comment by cramforce** [comment]
Since all content produced for Facebook must support muted autoplay, I don't really see much autoplay content not being designed for it.

Lets just do by default whatever Facebook does.

**Comment by aghassemi** [comment]
FB does autoplay with captions if captions are available* . They hide the subtitles when video is tapped and goes to full-screen and played with sound (since they are a native app, they are smarter about it and do that based on volume so full-screen videos may still has subtitles if volume on the device is on mute)

They also recently rolled out auto-captioning (for Ads only, not yet for individual videos) claiming videos with captions get 12% longer viewing time. https://adexchanger.com/digital-tv/facebook-adds-viewable-video-options-automated-captions-autoplay/

So I suggest with do:
- [ ] Enable this by default for <amp-youtube>
- [ ] Add opt-out option: `autoplay="withoutCaptions`
- [ ] Ensure auto-generated subtitles are not autoplayed

*To test, Go to the official Facebook page itself on Facebook, some of the videos on that page have captions (e.g. the one from Sept 7)

**Comment by cramforce** [comment]
Sounds great.

**Comment by abeck000** [comment]
ux notes re styling of captions: 

recommendations:
- change opacity of background color of caption box to 50%
- increase padding
- reduce the amount of text per caption shown
- only have one caption box, not two touching (currently)

cc @edlea-g

**Comment by aghassemi** [comment]
Blocked on several YT bugs. See b/34104933 for reference.

**Comment by ampprojectbot** [comment]
This issue hasn't been  updated in awhile. @wassgha Do you have any updates?

**Comment by ampprojectbot** [comment]
This issue hasn't been  updated in awhile. @wassgha Do you have any updates?

**Comment by ampprojectbot** [comment]
This issue hasn't been  updated in awhile. @wassgha Do you have any updates?

**Comment by aghassemi** [comment]
/to @alanorozco this is a good one, could be a good one for someone else (intern, new member) to do with your guidance too.

`amp-youtube`'s caption API wasn't great back then and it was only possible for `amp-video` it might had been fixed by now.

also interesting whether Stories would want this or not, my guess is not (@newmuis) so default behaviour may need to be different between normal AMP and stories.

**Comment by newmuis** [comment]
Yes, we would want this for stories.

**Comment by ampprojectbot** [comment]
This issue hasn't been  updated in awhile. @alanorozco Do you have any updates?
